compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaFunctionTypeSymbol.java

```diff
@@ -133,7 +133,12 @@ public BallerinaFunctionTypeSymbol(CompilerContext context, ModuleID moduleID,
     public Optional<TypeSymbol> returnTypeDescriptor() {
         if (returnType == null) {
             TypesFactory typesFactory = TypesFactory.getInstance(this.context);
-            this.returnType = typesFactory.getTypeDescriptor(this.typeSymbol.returnType);
+            if (this.getBType().getReturnType() != null) {
+                this.returnType = typesFactory.getTypeDescriptor(this.getBType().getReturnType());
+            } else {
+                this.returnType = typesFactory.getTypeDescriptor(this.typeSymbol.returnType);
```

Comment:
Shouldn't this be fixed in the beginning when we creating the `BallerinaFunctionTypeSymbol`? WDYT @pubudu91?

Reply:
What's the issue here? Is `typeSymbol.returnType` becoming `null` in some cases? If so, should look at why that's happening. The only case where it can be `null` is for the `function` typedesc. In all other cases, it should not be `null`.

Reply:
Yes, for `function` typedesc it is set as null. Let me check and update.

Reply:
@pubudu91 seems like the `BType` issue was previously mentioned in the **Remarks** section of a workaround fix at #28712

Reply:
@sanjana can't we fix the issue where we are getting the `null` for `typeSymbol.returnType` in this particular place?

Reply:
@dulajdilshan yes, I'm checking on it now. Seems like an issue in the TypeChecker.

Reply:
Seems like the `tSymbol.returnType` is being `null` by this stage. 

![image](https://user-images.githubusercontent.com/9464997/113828057-99c0a480-97a1-11eb-9c1a-c44758d3dfbe.png)

Reply:
Changed the approach.